error.go

```diff
@@ -156,6 +156,13 @@ func (e *Error) Cause() error {
 	return e.cause
 }
 
+// Unwrap returns the same value as Cause().
+// This method has been provided to allow errorx.Error to work with error.Is from
+// the Go standard library.
+func (e *Error) Unwrap() error {
```

Comment:
This is a welcome change, and even a but overdue by now. However, there are implications here that must be considered. 
`errors.Is()` inspects the entire errors chain. In errorx, on the other hand, there is a concept of opaque and transparent wrap. In both of those cases, some of errors in the chain are expected not to be observable programmatically. 
It seems to me that this logic must be supported in `Unwrap()` and `Is` methods. Possibly, it is better to implement `Is` and not to implement `Unwrap` at all. 

I'll think about this further, just sharing the considerations now.

Reply:
That's a good point, I'll try to take a stab at implementing `Is` instead in the next couple of days.

Reply:
@PeterIvanov From my reading of the code and the doc, my impression is that the only reason we want to make some wrappings opaque is because we want to set the type/traits/properties independently from the cause. But when an user is using `errors.Is`/`errors.As` they don't really care about those, so making `Unwrap` implementation to bypass the transparency check should be OK. But please correct me if I'm wrong.

My suggestion would be just implement `Unwrap` as an alias of `Cause`.  I don't think there's really a need to put it behind a build flag. The doc comment of it can just say it's provided to be compatible with go 1.13+ error wrapping APIs, and the direct use of it is still highly discouraged as `Cause`.

Reply:
I couldn't agree. 
Consider a wrapper over some library API. Library returns errors, but the wrapper aims to have a completely separate API in terms of errors. If errorx did not have an opaque wrap, this wrapper would have to transform original error into something that can't be handled like error - for example, via `err.Error()`. 
We don't want that. Wrapper should be used to wrap the original error in such a way that makes in visible only when it is printed into log - possibly, with its original stack trace.

Reply:
I think you might have a misunderstanding of the usages of `errors.Is`/`errors.As`.

When users are using `errors.Is`/`errors.As` to check against an error, they are looking to a specific error/type of error that they know they need to handle differently. For example they could use `if errors.Is(err, context.DeadlineExceeded) { return err }` to see if context deadline already passed, so there's no need to do any retries and we could just return early. If the underlying/wrapped error is some internal error that's not exported, there's no way `errors.Is`/`errors.As` would return true (because the user would have no way to construct the target), so that won't match anything. These kind of checks are only used against some common errors defined in the stdlib (e.g. context errors, `io.EOF`, `net.Error`, etc.). It's not meant to replace the current types/traits/properties defined in errorx package. It's a supplement to those.

Currently the default behavior of errorx wrapping is opaque, which could cause problem of some of the well defined stdlib errors are not check-able (again, the context example). The `Unwrap` implementation is meant to fix that problem.

Reply:
> the default behavior of errorx wrapping is opaque
Not quite. Wrapping of one error type with another is opaque, yes, but there is `errorx.Decorate()` and some other constructs like that, which are designed specifically for "no semantics change" kind of wrap. 

This is a controversial subject, of course. In some cases, an error like `context.DeadlineExceeded` is wrapped unconditionally, and you would like to inspect it anyway, and current errorx design discourages it. But the intent here is to provide isolation in many other cases, where undelying `EOF` or suchlike is semantically meaningful only to the immediate caller, and "leaking" such details to the outside world could create unwarranted expectations and future bugs. 

Errors are as much a part of package API as interfaces and types. Current `errorx` design aims to provide control over this API, as well.
In the future, if a good proposal is made as to how to defeat the unintentional incorrect opaque wrap in some cases, a breaking change could be made to implement it. But this change would have to affect existing `IsOfType` method, too, so it is a completely separate subject to what's done in this PR. For now, this impl should simply be consistent with the rest of the package.